### ISSUE TITLE
ConfigMap too long

### ISSUE THREAD

**Issue by macalinao** [jenkins-x/jenkins-x-platform#5222]
```
 % jx upgrade platform
Using provider 'aws' from team settings
using stable version 0.0.3832 from charts of jenkins-x/jenkins-x-platform from /home/igm/.jx/jenkins-x-versions
Cloning the Jenkins X cloud environments repo to /home/igm/.jx/cloud-environments
? A local Jenkins X cloud environments repository already exists, recreate with latest? No
Upgrading platform from version 0.0.3657 to version 0.0.3832
Creating /home/igm/.jx/adminSecrets.yaml from jx-install-config
Creating /home/igm/.jx/extraValues.yaml from jx-install-config
Using local value overrides file /home/igm/proj/abacusfi/infrastructure/kubernetes/jenkinsx/myvalues.yaml
error: unable to upgrade helm chart: failed to run 'helm upgrade --namespace jx --version 0.0.3832 --values /home/igm/.jx/cloud-environments/env-aws/myvalues.yaml --values /home/igm/.jx/extraValues.yaml --values /home/igm/.jx/adminSecrets.yaml --values /home/igm/.jx/cloud-environments/env-aws/secrets.yaml --values /home/igm/proj/abacusfi/infrastructure/kubernetes/jenkinsx/myvalues.yaml jenkins-x jenkins-x/jenkins-x-platform' command in directory '', output: '2019/04/08 04:04:20 Warning: Condition path 'jenkins.enabled' for chart jenkins returned non-bool value
2019/04/08 04:04:20 Warning: Condition path 'controllerbuild.enabled' for chart controllerbuild returned non-bool value
Error: UPGRADE FAILED: ConfigMap "jenkins-x.v3" is invalid: []: Too long: must have at most 1048576 characters'
```

Any ideas? Thanks!

**Comment by rawlingsj** [comment]
There's been a number of reports for this error on slack today, as yet I don't know what's changed or where this `jenkins-x.v3` configmap is coming from.  My guess is the error is slightly wrong and the issue is the "`.`" in the name `jenkins-x.v3`.  We may need to revert https://github.com/jenkins-x/jenkins-x-versions/commit/e8773579b4aa5eeb13e3897efb385a5b96cd4756.  @jstrachan does this configmap mean anything to you?  I can't seem to find what's creating it.

/priority critical-urgent
/area upgrade

**Comment by jnsvd** [comment]
I was involved in the Slack threads yesterday. We're trying to setup `jx` on Azure AKS. Right now we can only get it working with `--no-tiller=true`, but the goal is to use tiller.
We're testing with older versions of `jenkins-x-platform`, but we keep seeing the `ConfigMap` error in the tiller logs:

```
[storage/driver] 2019/04/09 10:46:26 create: failed to create: ConfigMap "jenkins-x.v1" is invalid: []: Too long: must have at most 1048576 characters
[tiller] 2019/04/09 10:46:26 warning: Failed to record release jenkins-x: ConfigMap "jenkins-x.v1" is invalid: []: Too long: must have at most 1048576 characters
```

Installation was done with command `jx install --git-private=true --environment-git-owner=<our-org> --version=0.0.3778`.

**Comment by jnsvd** [comment]
@rawlingsj We ran some more tests and concluded that version `0.0.3756` is the first version where we're seeing this error. In version `0.0.3737` this not an issue.

**Comment by macalinao** [comment]
Can confirm that running `jx upgrade platform --version=0.3737` fixes things for me.

**Comment by thienlh** [comment]
This issue still happening for me:

```
NAME               VERSION
jx                 2.0.3
jenkins x platform 0.0.3734
Kubernetes cluster v1.12.6-eks-d69f1b
kubectl            v1.14.0
helm client        v2.13.1+g618447c
helm server        v2.13.1+g618447c
git                git version 2.21.0
Operating System   Mac OS X 10.14.4 build 18E226
```
I tried to use
`jx upgrade platform --version=0.3737`
but another problem occurred 
```
Error: failed to download "jenkins-x/jenkins-x-platform" (hint: running `helm repo update` may help)
```

**Comment by jstrachan** [comment]
I've got a fix which is just going through BDD testing, will post as soon as its released to the version stream...

**Comment by n8mellis** [comment]
Is there a workaround for this latest issue with 'failed to download "jenkins-x/jenkins-x-platform"' issue?

**Comment by jstrachan** [comment]
should be fixed now

**Comment by thienlh** [comment]
Still happen for me.
```
NAME               VERSION
jx                 2.0.22
jenkins x platform 0.0.3734
Kubernetes cluster v1.12.6-eks-d69f1b
kubectl            v1.14.1
helm client        v2.13.1+g618447c
helm server        v2.13.1+g618447c
git                git version 2.21.0
Operating System   Mac OS X 10.14.4 build 18E226
```
`jx upgrade platform` logs
```
INFO[0032] Using provider 'eks' from team settings 
? A local Jenkins X versions repository already exists, pull the latest? Yes
INFO[0050] using stable version 2.0.78 from charts of jenkins-x/jenkins-x-platform from /Users/thienle/.jx/jenkins-x-versions 
INFO[0061] Upgrading platform from version 0.0.3734 to version 2.0.78 
INFO[0062] Cloning the Jenkins X cloud environments repo to /Users/thienle/.jx/cloud-environments 
? A local Jenkins X cloud environments repository already exists, recreate with latest? Yes
INFO[0063] Cloning the Jenkins X cloud environments repo to /Users/thienle/.jx/cloud-environments 
Enumerating objects: 1382, done.
Total 1382 (delta 0), reused 0 (delta 0), pack-reused 1382
INFO[0077] Creating /Users/thienle/.jx/adminSecrets.yaml from jx-install-config 
INFO[0077] Creating /Users/thienle/.jx/extraValues.yaml from jx-install-config 
INFO[0079] Deleting and cloning the Jenkins X versions repo 
INFO[0079] Cloning the Jenkins X versions repo https://github.com/jenkins-x/jenkins-x-versions.git with ref refs/heads/master to /Users/thienle/.jx/jenkins-x-versions 
Enumerating objects: 117, done.
Counting objects: 100% (117/117), done.
Compressing objects: 100% (57/57), done.
Total 1010 (delta 55), reused 108 (delta 51), pack-reused 893
error: unable to upgrade helm chart: failed to run 'helm upgrade --namespace jx --timeout 6000 --version 2.0.78 --values /Users/thienle/.jx/cloud-environments/env-eks/myvalues.yaml --values /Users/thienle/.jx/extraValues.yaml --values /Users/thienle/.jx/adminSecrets.yaml --values /Users/thienle/.jx/cloud-environments/env-eks/secrets.yaml jenkins-x jenkins-x/jenkins-x-platform' command in directory '', output: '2019/04/23 12:58:44 Warning: Condition path 'jenkins.enabled' for chart jenkins returned non-bool value
2019/04/23 12:58:44 Warning: Condition path 'controllerbuild.enabled' for chart controllerbuild returned non-bool value
UPGRADE FAILED
ROLLING BACK
Error: ConfigMap "jenkins-x.v30" is invalid: []: Too long: must have at most 1048576 characters
Error: UPGRADE FAILED: ConfigMap "jenkins-x.v30" is invalid: []: Too long: must have at most 1048576 characters'
```
Did I miss something here?

**Comment by jstrachan** [comment]
does `jx gc helm` help? you may have to remove the `jenkins-x ConfigMap` maybe to be able to upgrade?

**Comment by thienlh** [comment]
`jx gc helm` output
```
INFO[0019] Backing up configmaps/staging.v55.yaml.      
INFO[0019] Success.                                     
INFO[0020] ConfigMap staging.v55 deleted. 
```
Same output for `jx upgrade platform`

**Comment by jstrachan** [comment]
sorry I meant the `jenkins ConfigMap` - thats usually the big one in the older platform releases

**Comment by thienlh** [comment]
Is it safe to delete that ConfigMap? This is the production cluster so I just want to know the possible consequences.

**Comment by jstrachan** [comment]
you could always do `kubectl get cm jenkins -oyaml > backupcm.yaml` to back it up before removing it

**Comment by thienlh** [comment]
I backed up the ConfigMap, then deleted it. Run `jx gc helm` again and then `jx upgrade platform` but still same output.

**Comment by jstrachan** [comment]
managed to reproduce - I've hopefully a fix shortly once the BDD tests go green (added a new tiller bdd test to ensure it stays green too...)

**Comment by thienlh** [comment]
Thank you sir!

**Comment by jstrachan** [comment]
ok release is out which should fix this now with `jenkins-x/jenkins-x-platform` chart version `2.0.88`

**Comment by jstrachan** [comment]
I've added a BDD test for tiller to the [version stream](https://jenkins-x.io/architecture/version-stream/) to help keep this green going forwards

**Comment by thienlh** [comment]
Can confirm that the issue is fixed now.

**Comment by jstrachan** [comment]
@thienlh phew sorry about that! glad its working

**Comment by dimmel82** [comment]
It also worked for me in 2.0.88 (2.0.80 was still broken) !!!!!!!!!!!

**Comment by jstrachan** [comment]
@dimmel82 thanks! turns out there was a totally different issue ;) (the size of the platform chart that tiller uses to make an internal ConfigMap for tiller's own use)

**Comment by dimmel82** [comment]
Yes, I did some checks before googling for this error, and the "jenkins-x.vXX" config maps in the tiller's namespace (kube-system) were created until v35 but the creation of v36 failed with the error. When I base64 decoded the data of the jenkins-x.v35 cm and opened it as a zip archive I saw it was 622 KB (compressed size was 64KB).
It seems that the `jx upgrade` hit the 1 MB limit. After your fix the successfully created v36 has a size of 501 KB (compressed size is 53KB).
As I read on older issues of helm it seems to be a really hard limit, good to know that you found a way around.
The .helmignore file was that made the difference?
